tree.go

```diff
@@ -221,8 +221,8 @@ func (n *node) addRoute(path string, handlers HandlersChain) {
 				}
 
 				// Check if a child with the next path byte exists
-				for i := 0; i < len(n.indices); i++ {
-					if c == n.indices[i] {
+				for i, m := range []byte(n.indices) {
```

Comment:
Why not use the range over string as an Unicode codes? [EffectiveGO](https://golang.org/doc/effective_go.html#for)
```
for i, r := range n.indices {
```
It helps to us work with the unicode in a more clear way.

Reply:
just copy from https://github.com/julienschmidt/httprouter/blob/08a3b3d20bbeeab9ca4889bcdb65fec0423ef882/tree.go#L54-L63 ?

Reply:
In your example I see the same approach, as in current PR: `for i := range []byte(path) {`
I propose to use the range loop over `string`, not over `[]byte(string)`. That gives to us the correct Unicode processing.
We can fix it in next PR, just to be clear. Is there a reason why we don't use the range over `string`?
Thanks.